testcases/kernel/syscalls/sigaltstack/sigaltstack02.c

```diff
@@ -30,151 +21,45 @@
  * Expected Result:
  *  sigaltstack() should fail with return value -1 and set expected errno.
  *
- * Algorithm:
- *  Setup:
- *   Setup signal handling.
- *   Create temporary directory.
- *   Pause for SIGUSR1 if option specified.
- *
- *  Test:
- *   Loop if the proper options are given.
- *   Execute system call
- *   Check return code, if system call failed (return=-1)
- *	if errno set == expected errno
- *		Issue sys call fails with expected return value and errno.
- *	Otherwise,
- *		Issue sys call fails with unexpected errno.
- *   Otherwise,
- *	Issue sys call returns unexpected value.
- *
- *  Cleanup:
- *   Print errno log and/or timing stats if options given
- *   Delete the temporary directory(s)/file(s) created.
- *
- * Usage:  <for command-line>
- *  sigaltstack02 [-c n] [-e] [-f] [-i n] [-I x] [-p x] [-t]
- *	where,  -c n : Run n copies concurrently.
- *		-e   : Turn on errno logging.
- *		-f   : Turn off functionality Testing.
- *		-i n : Execute test n times.
- *		-I x : Execute test for x seconds.
- *		-P x : Pause for x seconds between iterations.
- *		-t   : Turn on syscall timing.
- *
- * History
- *	07/2001 John George
- *		-Ported
- *
- * Restrictions:
- *  This test should be executed by super-user (root) only.
  */
 
-#include <stdio.h>
-#include <sys/types.h>
-#include <unistd.h>
-#include <signal.h>
-#include <string.h>
-#include <ucontext.h>
-#include <errno.h>
-
-#include "test.h"
+#include <stdlib.h>
+#include "tst_test.h"
 
 #define INVAL_FLAGS	9999
 
-char *TCID = "sigaltstack02";
-int TST_TOTAL = 2;
-
-stack_t sigstk;			/* signal stack storing struct. */
-
-void setup(void);			/* Main setup function of test */
-void cleanup(void);			/* cleanup function for the test */
+static stack_t sigstk;			/* signal stack storing struct. */
 
-struct test_case_t {		/* test case struct. to hold diff. test.conds */
+static struct test_case {		/* test case struct. to hold diff. test.conds */
 	int flag;
 	int size;
 	char *desc;
 	int exp_errno;
-} Test_cases[] = {
-	{
-	INVAL_FLAGS, SIGSTKSZ, "Invalid Flag value", EINVAL},
+} tcases[] = {
+	{INVAL_FLAGS, SIGSTKSZ, "Invalid Flag value", EINVAL},
 	/* use value low enough for all kernel versions
 	 * avoid using MINSIGSTKSZ defined by glibc as it could be different
 	 * from the one in kernel ABI
 	 */
-	{
-	0, (2048 - 1), "alternate stack is < MINSIGSTKSZ", ENOMEM},
-	{
-	0, 0, NULL, 0}
-};
+	{0, (2048 - 1), "alternate stack is < MINSIGSTKSZ", ENOMEM} };
```

Comment:
We expect that `MINSIGSTKSZ` is not smaller than 2048. Looking on the values, it's true on all archs (sometimes is  higher 4096). I guess this can stay.